src/JWT/Builder/JwtBuilder.cs

```diff
@@ -276,7 +282,7 @@ public string Encode(Type payloadType, object payload)
             EnsureCanEncode();
 
             var dic = payloadType.GetProperties(BindingFlags.Instance | BindingFlags.Public)

```

Comment:
I still feel this is hacky. Is it possible to avoid doing it altogether?

Reply:
If we disallow using both .AddClaim()/.AddClaims() and ,Encode(payload) we should then be able to just send the payload directly to the serializer.

Reply:
Yes, that sounds like a reasonable approach

Reply:
I have added code to disallow mixing Encode(payload) with AddClaim/AddClaims